lib/duniverse.ml

```diff
@@ -165,23 +146,13 @@ module Repo = struct
     match urls with
     | [ (url, hashes) ] -> Ok { dir; url; hashes; provided_packages }
     | _ ->
-        (* If packages from the same repo were resolved to different URLs, we need to pick
-           a single one. Here we decided to go with the one associated with the package
-           that has the higher version. We need a better long term solution as this won't
-           play nicely with pins for instance.
-           The best solution here would be to use source trimming, so we can pull each individual
-           package to its own directory and strip out all the unrelated source code but we would
-           need dune to provide that feature. *)
-        let* highest_version_package =
-          Base.List.max_elt packages ~compare:(fun p p' ->
-              OpamPackage.Version.compare p.Package.opam.version p'.opam.version)
-          |> Base.Result.of_option
-               ~error:(Rresult.R.msg "No packages to compare, internal failure")
-        in
-        log_url_selection ~dev_repo ~packages ~highest_version_package;
-        let url = highest_version_package.url in
-        let hashes = highest_version_package.hashes in
-        Ok { dir; url; hashes; provided_packages }
+        let pp_hash = Fmt.of_to_string OpamHash.to_string in
+        Fmt.epr "XXX %a\n%!"
```

Comment:
I think it would be better to rework that logic under the assumption that there is only one such binding, anything else is a programming error.

Reply:
I've reworked the error message (and that XXX is a leftover, shouldn't have been there - thanks for spotting this)